api/v1beta1/nutanixmachine_types.go

```diff
@@ -99,6 +126,11 @@ type NutanixMachineSpec struct {
 	// +kubebuilder:validation:Required
 	SystemDiskSize resource.Quantity `json:"systemDiskSize"`
 
+	// dataDisks hold the list of data disks to be attached to the VM
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:listType=set
```

Comment:
does this have to be a set? i.e. I can't have two identical looking data disks on a machine?

Reply:
>  two identical looking data disks on a machine

Please, let's be more precise than "identical looking." This is a list of `[NutanixMachineVMDisk](https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/7593b7a86c05cb143324220833277abed5dbf204/api/v1beta1/nutanixmachine_types.go#L145)`, and the only required field for a `NutanixMachineVMDisk` is its size. The other fields are optional.

So, I guess the question is, if the list has two disks with the same size (and no other information), will validation reject that? What's a good way to test validation; will envtest work?

Reply:
> What's a good way to test validation; will envtest work?

Answering my own question: Yes, it will work. Faiq just added some validation for his templating changes, and I like them! https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/498/commits/1955dfcc3c42d1ca5f75bc205d675c24f05999ab